parquet/src/main/java/org/apache/iceberg/parquet/PruneColumns.java

```diff
@@ -49,7 +51,16 @@ public Type message(MessageType message, List<Type> fields) {
       Type field = fields.get(i);
       Integer fieldId = getId(originalField);
       if (fieldId != null && selectedIds.contains(fieldId)) {
-        builder.addField(originalField);
+        if (field != null) {
+          hasChange = true;
+          builder.addField(field);
+        } else {
+          if (isStruct(originalField)) {
+            builder.addField(originalField.asGroupType().withNewFields(Collections.emptyList()));
```

Comment:
Should `hasChange` be true here as well?

Reply:
Changing the ordering of the statements under (field != null) so that they match the usages here